### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by david-pj-rogers**
User model and authentication logic created

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,13 +15,18 @@ gem 'sidekiq', '~> 6.2.1'
 gem 'sidekiq-cron', '~> 1.2.0'
 gem 'sentry-ruby', '~> 4.6.4'
 gem 'sentry-rails', '~> 4.6.4'
+gem 'bcrypt'
+gem 'jsonapi.rb'
+gem 'jwt'
```
Comment (wmaciejak) [inline Gemfile:20]: Please add versions

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
-# frozen_string_literal: true
-
+# include Rack::Test::Methods
```
Comment (wmaciejak) [inline spec/spec_helper.rb:1]: I think that you can revert this changes

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -35,7 +36,7 @@
 RSpec.configure do |config|
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
-
+  # config.include Rack::Test::Methods
```
Comment (wmaciejak) [inline spec/rails_helper.rb:39]: Do you need it ?

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
 
+require 'rack/test'
```
Comment (wmaciejak) [inline spec/rails_helper.rb:3]: Why do you need it?

File: spec/controllers/api/v1/users/sessions_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        "Bearer #{JsonWebToken.encode(jti: User.last.jti.jti)}"
+    end
+
+    it 'ensures jti is destroyed' do
+      create(:user)
+      expect do
+        headers
+        destroy_request
+      end.to change(Jti, :count).by(-1)
+    end
```
Comment (wmaciejak) [inline spec/controllers/api/v1/users/sessions_controller_spec.rb:56]: What do you think about this version?
```ruby
  describe 'DELETE #destroy' do
    subject(:destroy_request) { delete :destroy, params: { id: user.id } }
    
    let(:user) { create(:user) }
    
     before do 
       request.headers['Authorization'] = "Bearer #{JsonWebToken.encode(jti: user.jti.jti)}"
     end 

    it 'ensures jti is destroyed' do
      expect { destroy_request }.to change(Jti, :count).by(-1)
    end
```

Comment (david-pj-rogers) [reply spec/controllers/api/v1/users/sessions_controller_spec.rb:56]: Much better 
Thank you

File: spec/rails_helper.rb (hunk 3 above)
Comment (david-pj-rogers) [reply spec/rails_helper.rb:39]: No

File: spec/controllers/api/v1/users/registrations_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,151 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::Users::RegistrationsController do
```
Comment (wmaciejak) [inline spec/controllers/api/v1/users/registrations_controller_spec.rb:3]: Try to rewrite those specs with `before {}` approach

**Review by wmaciejak — APPROVED** [review]
